src/github.js

```diff
@@ -1,91 +1,110 @@
 const { createOAuthDeviceAuth } = require('@octokit/auth-oauth-device');
-const { promisify } = require('util');
-const childProcess = require('child_process');
-const clipboard = require('clipboardy');
+const { request } = require('@octokit/request');
 
-const UI = require('./ui');
+const CLIENT_ID_PROD = 'ed7c193c5b64ee06192a';
 
-const exec = promisify(childProcess.exec);
-
-const CLIENT_ID = 'ed7c193c5b64ee06192a';
+const CLIENT_ID = process.env.DEV ? process.env.CLIENT_ID : CLIENT_ID_PROD;
 const CLIENT_TYPE = 'oauth-app';
-const SCOPE = ['delete_repo', 'repo'];
-const API_URL = 'https://api.github.com';
+const CLIENT_SCOPES = ['delete_repo', 'repo'];
 const API_PAGINATION = 100;
 
-async function auth() {
-  const spinner = UI.printAuthStart();
-
+async function auth(onVerificationCode) {
   const auth = createOAuthDeviceAuth({
     clientType: CLIENT_TYPE,
     clientId: CLIENT_ID,
-    scopes: SCOPE,
-    async onVerification(verification) {
-      UI.requestToken(spinner, verification);
-
-      clipboard.writeSync(verification.user_code);
-    },
+    scopes: CLIENT_SCOPES,
+    onVerification: onVerificationCode,
   });
 
   const { token } = await auth({ type: 'oauth' });
   setToken(token);
 
-  UI.printAuthFinished(spinner);
-
   return token;
 }
 
 async function getRepositories() {
-  const spinner = UI.printGetRepositoriesStart();
+  const res = await apiCall('GET', '/user/repos');
 
-  const curl = `curl ${getAuthHeader()} ${API_URL}/user/repos?per_page=${API_PAGINATION}`;
+  const scopes = res.headers['x-oauth-scopes'];
+  const repos = res.data;
 
-  const { stdout } = await exec(curl);
+  if (!checkPermissions(scopes)) throw new ScopesError();
```

Comment:
I think I'd put this check inside of `apiCall`, since you want that tested for both requests. It's unlikely that scopes might change between the two API calls but sometimes things happen in production...

https://twitter.com/FSD_in_6m/status/1400207129479352323